Redefines cyborg death, + some misc cyborg QoL/bugfixes

frawhst opened the pull request:
[cleanliness] [qol] [runtime] [bug]
## About the PR <!-- Describe the Pull Request here. What does it change? What other things could this impact? -->
The main bit of this PR is that cyborgs, upon death, will ALWAYS fall apart. However, the health of cyborg mobs no longer really matters. Death is defined as so:

1. Do we have a chest, a head, and a brain? We're alive.
2. Do we have a chest and a head, but no brain? We're a ghost, but the body is still alive.
3. Are we missing a chest or a head? We're dead. Fall apart.

To accommodate for this, all the fall-apart behavior now lives in death().

This also includes some other bugfixes:

- Removing a cyborg's arms functions properly, fixing #5681. A copy paste error that stood the test of time.
- Fixes a runtime when admin-gibbing something without a mind.
- PROBABLY fixes the issue where spontaneous intelligence creation cores wouldn't store their owner, resulting in a bricked brain if the head of a latejoin borg were destroyed. I located the issue (owner was only being set on the brain if it was ejected, not if it was dropped due to head destruction), but I can't really test if the fix works by myself, so I'm leaving the associated issues open for now.  

And some other QoL stuff:
- Latejoin brains now have some custom borg-themed examine text modeled after the warm/slimy/cold descriptions on regular brains, with extra text if the brain never had an inhabitant. Should help out roboticists a bit.
- Clearer message when a brain is vaporized due to silicon bannage.


## Why's this needed? <!-- Describe why you think this should be added to the game. -->
Cyborg death is... weird, and causes a lot of bugs. If you manage to die without falling apart, strange things can happen; roboticists can't interact with your body to take parts off; you might be existing without a chest; etc.

This removes that possibility entirely by simplifying things. Is the borg upright? They're alive! Are they a pile of parts? They're dead!

Fixes #4837. Fixes #4377. Fixes #4123. Fixes #4838. Fixes #4833. Fixes #4836. Fixes #4302.
(gross, but I just want github to link the issues correctly)
Likely solves #5277 and #4222, but requires further testing.

```changelog
(u)aloe
(*)Cyborgs die a bit differently now- the destruction of your head will always kill you, and any sort of death will cause you to fall apart. Report any new bugs, please!
```

mordent-goonstation commented on code/mob/living/silicon/robot.dm:
```diff
@@ -247,30 +246,26 @@
+			var/turf/T = get_turf(src)
+			for(var/obj/item/parts/robot_parts/R in src.contents)
+				R.set_loc(T)
+			src.part_chest = null
+			src.part_head = null
+			src.part_arm_l = null
+			src.part_arm_r = null
+			src.part_leg_l = null
+			src.part_leg_r = null
```
I don't *think* you need to `null` all of these here.

frawhst replied:
tested briefly, seems fine. Theoretically `qdel` should be clearing those out anyways.

mordent-goonstation commented on code/mob/living/silicon/robot.dm:
```diff
@@ -247,30 +246,26 @@
+		src.borg_death_alert()
+		if (!gibbed)
+			src.visible_message("<span class='alert'><b>[src]</b> falls apart into a pile of components!</span>")
+			var/turf/T = get_turf(src)
+			for(var/obj/item/parts/robot_parts/R in src.contents)
+				R.set_loc(T)
```
This is probably safer than using the same loc as the mob (which would let you have them gib in lockers), as I can definitely see some issues with e.g. gib in a pod putting all of the random bits within the pod and unretrievable.

Might cause the odd weird interaction here or there, but overall I'd prefer that over the edge-case bugs, so this seems fine.

frawhst replied:
Yeah, if the constant 'mechanics-deconstructing-shit-with-people-inside' issues have taught me anything, I'd rather err on the side of safety. I'd be open to using `loc` directly in the future, but it will require a lot of testing.

mordent-goonstation commented on code/mob/living/silicon/robot.dm:
```diff
@@ -1326,7 +1286,7 @@
 					src.compborg_force_unequip(3)
 					src.part_arm_r.set_loc(src.loc)
-					src.part_leg_r.holder = null
+					src.part_arm_r.holder = null
```
lol

mordent-goonstation commented:
Possibly affects #5973 ?

frawhst commented:
> Possibly affects #5973 ?
I don't think so? That seems like an issue of borging not caring if the person is in a critter (assuming the brain was formerly that person's- I can't think of any other way that would happen). I know the same thing can happen with changeling limbcritters being yanked out on borging/cloning, and I don't think I've touched anything that would fix it.

frawhst commented:
Removed all the null-setting on death, since `qdel` should handle that anyways. Also fixed an issue where shooting a dead limb would do no damage.
I started doing a general rework of the way cyborg limb damage/healing works to make it work a bit better with this borg death system, but that turned into a ton of changes, so I'm going to quit while I'm ahead and leave that for a future PR.
